Equipe 4 - Atividade 4 - Segunda Iteração

reueljonathan opened the pull request.

Commit:
Merge pull request #2 from pauloborba/master

Atualização projeto

Commit:
Adição da feature Parking Spots Reported By Users

Commit:
Adição da feature de lembrete de vaga.

Commit:
Adição da feature de preferência de destino.

Commit:
Adição da feature de preferência pessoal.

Commit:
Adição de feature de Realimentação de vaga

reueljonathan committed:
Merge pull request #1 from guedesbgeorge/master

Pull request para dar merge nas minhas features com as de Reuel.

Commit:
Cria a feature que sugere as vagas de estacionamento

Commit:
Melhora a descrição da feature 

Depois de melhor leitura do livro, acabei seguindo o modelo:
As [a kind of stakeholder]
So that [I can achieve some goal]
I wanto to [do some task]

Commit:
Merge pull request #3 from horaciojcfilho/master

Adição de feature de Realimentação de vaga

Commit:
Correção da feature Parking Spots Reported By Users

guedesbgeorge committed:
Merge pull request #1 from reueljonathan/master

Solicitação de merge

guedesbgeorge committed:
Merge pull request #2 from lperrusi/master

Solicitação de merge

Commit:
Corrigindo detalhes dos cenários

Commit:
Revisa o arquivo de cenário

Commit:
Create enviroment to use geb

Commit:
Pasta com chromedriver.exe

Commit:
Pasta test com cucumber atualizado com a pasta steps e pasta functional com steps e pages, adicao do gebconfig

guedesbgeorge committed:
Merge pull request #3 from lperrusi/master

Criação dos teste

Commit:
Files to run tests

Commit:
Adicionando steps de teste da feature de lembrete de vagas

Commit:
Merge pull request #4 from guedesbgeorge/master

Solicitação de Merge dos testes

Commit:
Adição de imports e correção de erros

Commit:
Adição de imports e correção de erros

Commit:
Add IDE files to gitgnore

Commit:
Remove IDE files to .gitignore work

Commit:
Remove .iml files from the git track

Commit:
Revert "Files to run tests"

This reverts commit 68c86babca42fdac4aed58ed8d00900d39248ae2.

Commit:
Recreate project from scratch

Commit:
Create some files to test the new configuration

Commit:
Implementacao das classes basicas Vaga e usuário

Commit:
Implementacao das classes basicas Vaga e usuário

Commit:
Alteração do GebConfig

Commit:
Correção de erros dos testes da feature Parking Spots Reported By Users

Commit:
Criação da pasta domain

Commit:
Criação das views da entidade usario

Commit:
Criação das view das entidade vaga

Commit:
Criação do controlador da entidade usuario

Commit:
Criação do controlador da entidade Vaga

Commit:
Implementando feature de lembrete de vagas

Commit:
Create views and controllers for usuario and vaga

Commit:
merge with master

Commit:
Change gitignore

Commit:
Change gitignore, merge GebConfig

Commit:
merge with new-configuration

Commit:
Implementação da feature de Reserva de vagas

Commit:
Resolvendo conflitos

Commit:
Resolvendo conflitos do merge de George

Commit:
Merge with master

Commit:
implementa feature de busca de vaga por setor

Commit:
Adição do getDistance

Commit:
Adição dos atributos de localização, x e y

Commit:
Update GebConfig.groovy

Commit:
Update UsuarioController.groovy

Commit:
Merge com master de horacio

Commit:
Update VagaController.groovy

Commit:
implementação de feature de busca por setor

Commit:
Update Usuario.groovy

Commit:
Update Vaga.groovy

Commit:
Update _form.gsp

Commit:
Update create.gsp

Commit:
Update edit.gsp

Commit:
Update index.gsp

Commit:
Update show.gsp

Commit:
Update _form.gsp

Commit:
Update create.gsp

Commit:
Update edit.gsp

Commit:
Update index.gsp

Commit:
Update show.gsp

Commit:
Resolvendo conflitos de Perrusi

Commit:
Merge branch 'master' of https://github.com/horaciojcfilho/SistemaDeVagasDeEstacionamento

Commit:
Resolvendo de vez conflitos de Perrusi

Commit:
Merge branch 'master' of https://github.com/horaciojcfilho/SistemaDeVagasDeEstacionamento

Commit:
Implementacao de feature

Commit:
Implementacao de feature

Commit:
Merge pull request #3 from reueljonathan/master

merge de feature com master de horácio

Commit:
Removendo arquivos desnecessários

Commit:
Clear repository

Commit:
Create .gitignore

Commit:
Update .gitignore

Commit:
Create readme.md

Commit:
Initial version of the project

Commit:
Update README

Commit:
Update gitignore, add gebconfig

Commit:
Corrige configuração do projeto

Commit:
Merge branch 'master' of github.com:reueljonathan/SistemaDeVagasDeEstacionamento

Commit:
Remove arquivos não desejados do projeto

Commit:
Remove arquivos não desejados remanescentes

Commit:
Merge branch 'master' of github.com:reueljonathan/SistemaDeVagasDeEstacionamento

Commit:
Limpa meu repositório

Commit:
Sobe os novos arquivos do projeto com configuração corrigida

Commit:
Começa a implementação da sugestão de vaga

Commit:
Corrige problema com os assets

Commit:
Prepara os testes do Cucumber

Commit:
Avança na implementação do cenário

Commit:
Avança mais na implementação

Commit:
Mais avanço na implementação da feature

Commit:
Adicionar o Apache Shiro

Commit:
Devolve as capacidades de usuário

Commit:
Adiciona os desenvolvedores como usuários-padrão para evitar login

Commit:
Atualiza com a última versão feita

Commit:
Repositorio com os testes funcionando.

GRACAS A DEUS

Commit:
Removendo arquivios desnecessarios

Commit:
Começa a implementação do teste

Commit:
Corrige a implementação do teste através da negociação de conteúdo

Commit:
Firulando

Commit:
Limpando repositório

Commit:
Adicionando arquivos atualizados do projeto

Commit:
Modificando tela pós-login

Commit:
Merge com fork de george

Commit:
Create first tests for the book feature

Commit:
Implementação do segundo teste e melhorias de código

Commit:
Devolve o arquivo do Travis CI

Commit:
Update .travis.yml

Commit:
Somente o Firefox é instalado por padrão no Linux

Commit:
Implementação da feature de lembrete de vagas após adicionar o shiro

Commit:
Tirando informações desnecessárias ao criar vaga

Commit:
Adicionando arquivo de feature que tinha sido removido ao limpar o repositório

Commit:
Atualizando arquivo de feature

Commit:
Merge com fork de horácio

Commit:
Merge com fork de horácio

Commit:
Implementação do teste de controlador de lembrete de vaga

Commit:
Termina teste de gui e controlador

Commit:
Esforço para tentar atualizar os plugins usados para versões mais atuais

Resolvendo problemas com o Firefox

Commit:
Atualização do projeto

Commit:
Atualização do gitignore

Commit:
Estranhamente os Chrome drivers se transformaram em arquivos de texto

Possivelmente por conta da política de privacidade do Shiro

Commit:
Merge com reuel

Commit:
Adição de teste de GUI

Commit:
Minor fixes

Commit:
Merge pull request #5 from lperrusi/new-configuration

Merge new-configuration

reueljonathan committed:
Merge pull request #3 from lperrusi/new-configuration

Merge da atividade 4

Commit:
Corrigindo implementação de feature

Commit:
Tá na hora

Commit:
Iniciando implementação de teste de GUI

Commit:
Faltando implementar teste de GUI

Commit:
Merge resolvendo conflito

Commit:
Terminando teste de GUI

reueljonathan committed:
Merge pull request #4 from guedesbgeorge/master

Merge para atividade 4

Commit:
Grails sendo Grails

Commit:
Testes de controlador funcionando

Commit:
Corrige a implementação da feature

Commit:
Feature quase pronta

Commit:
Tenta corrigir comportamento estranho da feature

lperrusi commented on test/cucumber/ParkingSpaceBooking.feature:
```diff
@@ -0,0 +1,19 @@
+    Given I am active with login "reuel.jonathan" and password "123"
+      And I am at the parking space list page
+      And I see the parking space "CIN-02" available in the list
+     When I ask to book the parking space "CIN-02" available
```
no passo anterior já fala que a vaga esta livre

lperrusi commented on test/cucumber/steps/ParkingSpaceBookingSteps.groovy:
```diff
@@ -0,0 +1,83 @@
+    parkingSpace.setDescription(description)
+    parkingSpace.setSector("CIn")
+    controller.saveParkingSpace(parkingSpace)
+    parkingSpaceDescription = description
```
Criação da vaga deveria ser feita em CreateParkingSpaceTestAndAplication

lperrusi commented on test/cucumber/steps/ParkingSpaceSuggestionSteps.groovy:
```diff
@@ -0,0 +1,80 @@
+
+And(~/^the parking space "(.*?)" is from the "(.*?)" sector$/) { String description, String sector ->
+    def controller = new ParkingSpaceController()
+    controller.save(new ParkingSpace([description: description, sector: sector]))
```
Criação da vaga deveria ser feita em CreateParkingSpaceTestAndAplication

Commit:
Teste de GUI quase pronto

lperrusi commented on test/cucumber/steps/ParkingSpaceSuggestionSteps.groovy:
```diff
@@ -0,0 +1,80 @@
+Then(~/^the systems does not inform the parking space "(.*?)" to the user$/) { String description ->
+    def response = parkingSpaceController.response.json
+
+    def parkingSpace = response.find { it.description == description }
```
find deveria ser executado no controlador, não no teste

Commit:
Teste de GUI funcionando

guedesbgeorge committed:
Update ParkingSpaceReminder.feature

Commit:
Implementa o segundo teste de GUI

Commit:
Faz os testes rodarem no Firefox

Commit:
O Firefox não tava gostando muito do Shiro

Commit:
Aplica refatoração Extract Method

Remove muitas duplicações de código causadas por dificuldades técnicas com a linguagem Groovy

Commit:
Remove depedências desnecessária e melhora o README

Commit:
Marionette parece gostar mais dessa abordagem para verificar se um elemento existe na página

Evita condições de corrida nas checagens de existência de um elemento na página

Commit:
Criando novo teste de controlador

Commit:
Implementando novo cenário de GUI

Commit:
Melhora as páginas HTML

Commit:
Implementa mais um teste de GUI

Commit:
Mais um teste de GUI

Versão GUI do teste de controlador

Commit:
Mais um teste de GUI

Aparentemente o Marionette não funciona bem em versões do Firefox outras que não a Developer, Beta e Nightly

Commit:
Torna a página de Sign up mais amigável para outros WebDrivers (Firefox e Edge)

Commit:
Melhora a página de sugestões de vaga para evitar reloading e testes rodarem no Firefox

Commit:
Mais um teste de GUI

Commit:
Mais 2 testes de GUI

Commit:
Aplica refatoração Extract Method e modernização do código

Usa lambdas ao invés da maneira antiga de Java para implementar interfaces

Commit:
Usa sintaxe mais moderna de Grails

Commit:
Remove constraint inválido

Commit:
Aplica refatoração Simplify Conditional Expression

Commit:
Reescreve a ação 'register' visando remover warnings reportados pela análise estática do código

Commit:
Reverte "Remove constraint inválido"

A ferramenta tinha acusado que o constraint era inválido e que deveria ser removido mas foi apenas um falso-positivo.

Commit:
Pequenas melhorias no README

Commit:
Pequeno deslize na escrita do README

Commit:
Adicionando mais dois cenários

Commit:
Merge branch 'master' of https://github.com/guedesbgeorge/SistemaDeVagasDeEstacionamento into merge

Commit:
Retorna para o Grails 2.4.3

Para evitar problemas durante a integração do código

Commit:
Resolvendo conflitos

reueljonathan committed:
Merge pull request #6 from horaciojcfilho/sugestao-vaga

Implementação da funcionalidade de sugestão de vaga

Commit:
Aplicando refatoração de código sugeridas pela ferramenta do IntelliJ

reueljonathan committed:
Merge pull request #7 from guedesbgeorge/master

Criação de dois novos cenários e aplicando refatoração de código

Commit:
Adição de teste de GUI e remoção de codigo duplicado

Commit:
Fix controller method

Commit:
Correção dos testes

Commit:
Merge branch 'master' of https://github.com/reueljonathan/SistemaDeVagasDeEstacionamento

Commit:
Merge branch 'new-configuration' of https://github.com/lperrusi/SistemaDeVagasDeEstacionamento

Commit:
merge de testes

Commit:
Merge pull request #6 from reueljonathan/master

Equipe 4 - Atividade 4 - Segunda Iteração

reueljonathan committed:
Merge pull request #8 from lperrusi/new-configuration

Equipe 4 - Atividade 4 - Segunda Iteração

Commit:
Refactoring e melhoria de trechos de código

Commit:
Merge com repositorio principal

Commit:
Removendo arquivo duplicado

Commit:
Adicionando novamente arquivo que estava duplicado

Commit:
Instala chrome para poder usar chromedriver nos testes de GUI

Commit:
Instala chrome para poder usar chromedriver nos testes de GUI

Commit:
Instala chrome para poder usar chromedriver nos testes de GUI 3

Commit:
Volta a tentar testar com o firefox

Commit:
Testa com chrome e ubuntu 14 mais uma vez

Commit:
Update .travis.yml

Commit:
Update .travis.yml

Tenta mudar a versão do Firefox para rodar os testes

Commit:
Update .travis.yml

Tenta partir para uma versão mais consolidada.

Commit:
Passa a aceitar um chrome driver sendo passado como parâmetro

Commit:
Nova abordagem para o chrome driver

Commit:
Internacionaliza os testes

Commit:
Internacionaliza uma parte do código que tinha sido esquecida

Commit:
Usa a versão do Chrome já instalada no Travis

Commit:
Update .travis.yml

Commit:
Update .travis.yml

Commit:
Update .travis.yml

Commit:
Update .travis.yml

Commit:
Update .travis.yml

Commit:
Update .travis.yml

pauloborba commented on test/cucumber/steps/ParkingSpotsReportedByUsersSteps.groovy:
```diff
@@ -2,56 +2,65 @@ package steps
+//Given that the system indicated to the user "Pedro" the parking spot "20"
+
+When(~'^the user "([^"]*)" fills in the system that he parked in the parking spot "([^"]*)"$'){ String user, int numSpot ->
+
```
?
